nclib/netcat.py

```diff
@@ -683,21 +683,17 @@ def interact(self, insock=sys.stdin, outsock=sys.stdout):
                 outsock.write(self.buf)
                 outsock.flush()
                 self.buf = b''
-            dropped = False
-            while not dropped:
-                r, _, _ = _select.select([self.sock, insock], [], [])
-                for s in r:
-                    if s == self.sock:
-                        a = self.recv(timeout=None)
-                        if a == b'':
-                            dropped = True
-                        else:
-                            outsock.write(a)
-                            outsock.flush()
-                    else:
-                        b = os.read(insock.fileno(), 4096)
-                        self.send(b)
-            raise NetcatError("Connection dropped!")
+
+            while True:
+                readable_socks, _, _ = select.select([self.sock, insock], [], [])
+                if self.sock in readable_socks:
+                    data = self.recv(timeout=None)
+                    if not data:
+                        raise NetcatError
```

Comment:
Can this have a message?

Reply:
It's caught immediately and hidden from user anyway...